### PR TITLE
Migrate 3 modules to JUnit

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

File: concurrent/src/test/java/io/airlift/concurrent/TestAsyncSemaphore.java [hunk 1]
```diff
@@ -289,122 +288,135 @@ public void testNoStackOverflow()
...
-        future.set("value");
-        assertThat(result).isDone();
-        assertThat(result.get()).isEqualTo(ImmutableList.of("value"));
+        for (int concurrency : concurrency()) {
+            SettableFuture<String> future = SettableFuture.create();
+            ListenableFuture<List<String>> result = AsyncSemaphore.processAll(ImmutableList.of(1), (i) -> future, concurrency, directExecutor());
+            assertThat(result).isNotDone();
+            future.set("value");
+            assertThat(result).isDone();
+            assertThat(result.get()).isEqualTo(ImmutableList.of("value"));
```
Comment (findepi) [inline concurrent/src/test/java/io/airlift/concurrent/TestAsyncSemaphore.java:317]: assertions should include value of `concurrency` 
(but i'd recommend using junit builtin support for parameters provider instead)

File: concurrent/src/test/java/io/airlift/concurrent/TestMoreFutures.java [hunk 2]
```diff
@@ -815,7 +810,7 @@ private static void assertFailure(Thrower thrower, Consumer<Throwable> verifier)
             verifier.accept(throwable);
             return;
         }
-        fail("expected exception to be thrown");
+        abort("expected exception to be thrown");
```
Comment (findepi) [inline concurrent/src/test/java/io/airlift/concurrent/TestMoreFutures.java:813]: this was a failure. is `abort` a failure or "skip exception"?

(same in other places)

File: configuration/src/test/java/io/airlift/configuration/testing/TestConfigAssertions.java [hunk 3]
```diff
@@ -392,7 +391,7 @@ public void testDeprecatedPropertiesFailUntestedProperties()
         }
 
         if (pass) {
-            fail("Expected AssertionError");
+            abort("Expected AssertionError");
```
Comment (findepi) [inline configuration/src/test/java/io/airlift/configuration/testing/TestConfigAssertions.java:394]: fail?

**Comment by wendigo** [comment]
Replaced by https://github.com/airlift/airlift/pull/1195

**Pull request closed without merging**